packages/playwright-core/src/server/dispatchers/pageDispatcher.ts

```diff
@@ -291,19 +296,20 @@ export class PageDispatcher extends Dispatcher<Page, channels.PageChannel> imple
 }
 
 
-export class WorkerDispatcher extends Dispatcher<Worker, channels.WorkerChannel> implements channels.WorkerChannel {
-  static fromNullable(scope: DispatcherScope, worker: Worker | null): WorkerDispatcher | undefined {
+export class WorkerDispatcher extends Dispatcher<Worker, channels.WorkerChannel, PageDispatcher | BrowserContextDispatcher, WorkerDispatcher> implements channels.WorkerChannel {
+  _type_Worker = true;
+
+  static fromNullable(scope: PageDispatcher | BrowserContextDispatcher, worker: Worker | null): WorkerDispatcher | undefined {
     if (!worker)
       return undefined;
     const result = existingDispatcher<WorkerDispatcher>(worker);
     return result || new WorkerDispatcher(scope, worker);
   }
 
-  _type_Worker = true;
-  constructor(scope: DispatcherScope, worker: Worker) {
+  constructor(scope: PageDispatcher | BrowserContextDispatcher, worker: Worker) {
     super(scope, worker, 'Worker', {
       url: worker.url()
-    });
+    }, true);
```

Comment:
It seems like we can declare every object a scope now. Do we have an example where it's not a scope and it matters?

Reply:
I might be able to do that in a follow-up.